core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -1671,6 +1677,104 @@ private IntersectOnCalcsToIntersectUnifyRule() {
     }
   }
 
+  /**
+   * A {@link SubstitutionVisitor.UnifyRule} that matches a
+   * {@link MutableSort} to a {@link MutableSort} where the query and target
+   * have the same inputs but might not have the same order.
+   */
+  private static class SortToSortUnifyRule extends AbstractUnifyRule {
+
+    public static final SortToSortUnifyRule INSTANCE = new SortToSortUnifyRule();
+
+    private SortToSortUnifyRule() {
+      super(any(MutableSort.class), any(MutableSort.class), 0);
+    }
+
+    @Override protected @Nullable UnifyResult apply(UnifyRuleCall call) {
+      final MutableSort query = (MutableSort) call.query;
+      final MutableSort target = (MutableSort) call.target;
+      final RelCollation queryCollation = query.collation;
+      final RelCollation targetCollation = target.collation;
```

Comment:
Is it possible to simplify the inclusion relationship of 'RelCollation'? 
Maybe it can be extracted into a method.

Reply:
OK, Got it.